metricflow/test/model/validations/test_data_warehouse_tasks.py

```diff
@@ -98,6 +99,52 @@ def test_validate_data_sources(  # noqa: D
     assert "Unable to access data source `test_data_source2`" in issues[0].message
 
 
+def test_build_dimension_tasks(data_warehouse_validation_model: UserConfiguredModel) -> None:  # noqa: D
+    tasks = DataWarehouseTaskBuilder.gen_dimension_tasks(model=data_warehouse_validation_model)
+    assert len(tasks) == 1  # on data source query with all dimensions
+    assert len(tasks[0].on_fail_subtasks) == 2  # a sub task for each dimension on the data source
+    assert (
+        "SELECT (ds) AS col0, (is_dog) AS col1 FROM (SELECT true AS is_dog, '2022-06-01' AS ds) WHERE is_dog IS NOT NULL"
```

Comment:
This will need to be updated after you rebase.

Reply:
Updated 👍